### PR TITLE
Fix Mining Issues [in progress]

### PR DESCRIPTION
**Pull request by WizardJesse1**
* Fix ore message when you successfully got an ore.
* Fix issue #247 and currently redid how graham had it originally

### CODE REVIEW

File: data/plugins/skill/mining/mining.rb [hunk 1]
```diff
@@ -21,12 +21,7 @@ def initialize(mob, position, ore)
   end
 
   def find_pickaxe
-    weapon = mob.equipment.get(EquipmentConstants::WEAPON)
-    PICKAXE_IDS.each do |id|
-      return PICKAXES[id] if (!weapon.nil? && weapon.id == id) || mob.inventory.contains(id)
-    end
-
-    nil
+    PICKAXES[PICKAXE_IDS.select {|id| PICKAXES[id].level <= mob.skill_set.get_skill(Skill::MINING).current_level && (mob.inventory.contains(id) || mob.equipment.contains(id))  }.max]
```
Comment (garyttierney) [inline data/plugins/skill/mining/mining.rb]: Could you break this condition down into multiple lines? It spans too many columns.

**Comment by Major-** [comment]
```ruby
def find_pickaxe
  level = mob.skill_set.get_skill(Skill::MINING).current_level
  inventory, equipment = mob.inventory, mob.equipment

  indices = PICKAXE_IDS.select do |id| 
    PICKAXES[id].level <= level && (inventory.contains(id) || equipment.contains(id))
  end

  PICKAXES[indices.max]
end
```

Still pretty hacky but that will do for now

**Comment by cubeee** [comment]
current_level should be changed to maximum_level since you can't boost your level to use better pickaxes

**Comment by WizardJesse1** [comment]
@cubeee true but what about when u get stat reduced on mining does it affect ur pickaxes?

**Comment by cubeee** [comment]
@WizardJesse1 No, it just lowers your chances of getting ore

**Comment by WizardJesse1** [comment]
o cool okay thanks

**Comment by Major-** [comment]
With this branch open (and assuming you haven't made any further commits to it, or **any** changes to other files), run:

```
git reset --hard HEAD~1
git fetch origin master
git rebase origin/master
```

**Pull request closed without merging**